runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6846,6 +6921,37 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
                (TR::Options::getCmdLineOptions()->getOption(TR_EnableJITaaSHeuristics) || localColdCompilations))
                doLocalCompilation = true;
 
+#if DEV_DBG_ENABLE_CHANGE
+            // If it is a remote sync compilation, change it to a local sync compilation.
+            // After the local compilation is completed, a remote async compilation will be scheduled
+            if (!doLocalCompilation &&
+                !entry->_async &&
+                !entry->_changedToLocalSyncComp &&
+                TR::Options::getCmdLineOptions()->getOption(TR_EnableJITaaSHeuristics) &&
+                !TR::CompilationInfo::isCompiled(method) &&
+                !TR::Options::getCmdLineOptions()->getOption(TR_DontDowngradeToCold) &&
```

Comment:
Replace the test on `TR_DontDowngradeToCold` with a test on opt level. If the optLevel is 'warm' or higher then we don't to do this optimization. In theory the test on optlevel is not even required because of the piece that transforms cheap compilations into local ones, but the code becomes more explicit that way.